src/app/researcher/study/[encodedStudyId]/edit/form.tsx

```diff
@@ -1,84 +1,130 @@
 'use client'
 
-import React from 'react'
+import React, { useState } from 'react'
 
 import { Form as HookForm, useForm } from 'react-hook-form'
-import { Checkbox, Select, Textarea, TextInput } from 'react-hook-form-mantine'
-
+import { Checkbox, Textarea, TextInput } from 'react-hook-form-mantine'
 import { inputStyle, labelStyle } from './style.css'
-import { Button, Flex, Stack, Text } from '@mantine/core'
+import { Button, Flex, Group, Stack, Text } from '@mantine/core'
 import { onUpdateStudyAction } from './actions'
 import { FormValues, schema, zodResolver } from './schema'
+import Link from 'next/link'
 
 export const Form: React.FC<{ studyId: string; study: FormValues }> = ({ studyId, study }) => {
-    const { control } = useForm<FormValues>({
+    const {
+        control,
+        // formState: { isValid },
+        getValues,
+    } = useForm<FormValues>({
         resolver: zodResolver(schema),
         defaultValues: study,
+        mode: 'onChange',
     })
+    const [isDisabled, setIsDisabled] = useState(true)
+    const [isFormValid, setIsFormValid] = useState(true)
+    const enableButton = () => {
+        setIsDisabled(false)
+    }
+    const checkFormValid = () => {
```

Comment:
I started digging into "why" you might be needing this - could be related to a spelling error in `schema.ts` in the `refine` callback